cedar-lean/Cedar/Thm/Validation/Typechecker/Call.lean

```diff
@@ -373,6 +373,46 @@ theorem type_of_call_ipAddr_recognizer_is_sound {xfn : ExtFun} {xs : List Expr}
       apply bool_is_instance_of_anyBool
     }
 
+-- It's impossible to call `unknown` in a well typed way
+theorem type_of_unknown_call_untyped :
+  ∀ (f : ExtFun) (xs : List Expr) (c : Capabilities) (env : Environment),
+  f = .unknown ∨ f = .error →
+  ∃ e, typeOf (.call f xs) c env = err e :=
+  by
+  intros f xs c env h
+  unfold typeOf
+  simp
```

Comment:
Our recommendation in one of our READMEs is to avoid bare `simp` except to close a goal, for proof stability reasons (hedging against lemmas being added/removed from `simp` in future versions of Lean).  You can write `simp?` and then click to accept the suggestion of an equivalent `simp only` command.